superset/assets/src/dashboard/util/getFilterScopeNodesTree.js

```diff
@@ -25,83 +27,93 @@ import {
 
 const FILTER_SCOPE_CONTAINER_TYPES = [TAB_TYPE, DASHBOARD_ROOT_TYPE];
 
-export default function getFilterScopeNodesTree({
+function traverse({
+  currentNode = {},
   components = {},
-  isSingleEditMode = true,
-  checkedFilterFields = [],
-  selectedChartId,
+  filterFields = [],
+  selectedChartId = 0,
 }) {
-  function traverse(currentNode) {
-    if (!currentNode) {
-      return null;
-    }
-
-    const type = currentNode.type;
-    if (CHART_TYPE === type && currentNode.meta.chartId) {
-      const chartNode = {
-        value: currentNode.meta.chartId,
-        label:
-          currentNode.meta.sliceName || `${type} ${currentNode.meta.chartId}`,
-        type,
-        showCheckbox: selectedChartId !== currentNode.meta.chartId,
-      };
-
-      if (isSingleEditMode) {
-        return chartNode;
-      }
+  if (!currentNode) {
+    return null;
+  }
 
-      return {
-        ...chartNode,
-        children: checkedFilterFields.map(filterField => ({
-          value: `${currentNode.meta.chartId}:${filterField}`,
-          label: `${currentNode.meta.chartId}:${filterField}`,
-          type: 'filter_box',
-          showCheckbox: false,
-        })),
-      };
-    }
+  const type = currentNode.type;
+  if (CHART_TYPE === type && currentNode.meta.chartId) {
```

Comment:
this will throw an error if currentNode isn't defined (as `{}.meta.chartId` will throw)

Reply:
added extra check.